client/signup/controller.js

```diff
@@ -323,6 +320,17 @@ export default {
 		} );
 
 		next();
+
+		// If the flow has siteId or siteSlug as query dependencies, we should not clear selected site id
+		if (
+			! providesDependenciesInQuery?.includes( 'siteId' ) &&
+			! providesDependenciesInQuery?.includes( 'siteSlug' ) &&
+			! isManageSiteFlow
+		) {
+			// We do this after `next()`; otherwise, components that are currently rendered and depend
```

Comment:
Moving this block until after the route has changed stops currently rendered components that depend on `state.ui.selectedSiteId` from breaking.

Reply:
I think we should include this eaead4affbee0ae6079246fa4cc81db23c2ea419 in this PR too. It just moves the state update to avoid errors.